spinalcordtoolbox/scripts/sct_image.py

```diff
@@ -324,6 +330,39 @@ def main(argv=None):
         dim = dim_list.index(dim)
         im_out = split_data(im_in, dim)
 
+    elif arguments.stitch is not None:
+
+        # preserve original orientation (we assume it's consistent among all images)
+        orig_ornt = im_in_list[0].orientation
+
+        # preserve current directory
+        curdir = os.getcwd()
+
+        # creating a temporary folder in which all temporary files will be placed and deleted afterwards
+        path_tmp = tmp_create(basename="image-stitching")
+
+        fnames_in = []
+
+        # copy all files to temporary, reorient and place in temp directory
+        for file in arguments.i:
+            file_name = os.path.basename(file)
+            temp_file_path = os.path.join(path_tmp, file_name)
+            print(temp_file_path)
+            copy(file, temp_file_path, verbose='verbose')
+            im_in = Image(temp_file_path)
+            im_out = change_orientation(im_in, 'RPI')
+            im_out.save(temp_file_path, dtype=output_type, verbose=verbose)
+            # display_viewer_syntax([temp_file_path], verbose=verbose)
+            fnames_in.append(file_name)
+
+        os.chdir(path_tmp)
+
+        # stitch and reorient
+        im_out = stitch_images(fnames_in=fnames_in)
```

Comment:
> Potentially I would also like to implement a quality check for this feature (probably a snapshot of the stitched image), should we tackle this in the same PR or move the related discussion to another PR/issue?

I think it would be good to discuss it here, because how we add a QC report will have implications for how we organize this PR:

- `sct_image -stitch`: Since `sct_image` has no QC reporting, we would have to think about whether to support QC reporting for just `sct_image -stitch`, or whether we would make it available to any `sct_image` command that outputs a single 3D image, or if we would want to try to accommodate 4D output/lists of images/etc.
- Alternatively, we could reconsider moving the logic to a separate `sct_stitch` command, and add QC specifically for `stitching` this way. (That way, we don't have to worry about managing QC reports for other `sct_image` functions. But, perhaps this is overkill for simply displaying a single output image.)

Reply:
Looking into `sct_image`, I feel there is no need to provide quality checks for other functions other than the `--stitch` option :slightly_smiling_face:, so perhaps `sct_stitch` would be a nice option then again?

Moreover, I am not 100% sure how we want to implement the quality check for stitching as well.
In general, we will always have 2-3 images and the resulting image. For reference, let's assume we have:


input:
```
sub-123456_ses-20220101_acq-ax_chunk-1_T2w.nii.gz
sub-123456_ses-20220101_acq-ax_chunk-2_T2w.nii.gz
sub-123456_ses-20220101_acq-ax_chunk-3_T2w.nii.gz
```
output:
```
sub-123456_ses-20220101_acq-ax_T2w.nii.gz
```
Then it would be probably nice for the user(s) to have screenshots of all chunks and the resulting image side-by-side to evaluate whether stitching worked (or not)? Or what would be a practical way to analyze this? I feel it's not sufficient to only show the resulting image, as the user might miss that the 3rd chunk was successfully stitched to the upper two chunks (I believe this is important, as stitching still outputs an image when the registration fails, e.g. when the images are oriented in the wrong convention).

Reply:
Small heads-up: I'm working on this now, and have a rough prototype for the QC reporting:

![qc_report_demo](https://user-images.githubusercontent.com/16181459/190446333-efe59ce0-d656-4360-a842-38bb049d7de0.gif)

(You can toggle between the unstitched images and the stitched image using the right arrow.)

I just need to do a bit of cleanup and I'll push the changes. :slightly_smiling_face:

Reply:
My apologies for the delay! I've added the QC report in https://github.com/spinalcordtoolbox/spinalcordtoolbox/pull/3865/commits/02102c18eea7e7821e47398bbbe918794185fa8a.

I needed to do a bit of extra cleanup in order to have the right images available for the QC report, though. So, there are 14 extra commits that make small tweaks here and there. :slightly_smiling_face: 

I'm going to self-review to raise some additional questions about the design of the QC report, too.